packages/core/src/deal-tracker/spade-oracle-sync-tick.js

```diff
@@ -0,0 +1,238 @@
+// @ts-expect-error no types available
+import { ZSTDDecompress } from 'simple-zstd'
+import { Readable } from 'stream'
+// @ts-expect-error no types available
+import streamReadAll from 'stream-read-all'
+import { encode, decode } from '@ipld/dag-json'
+import { RecordNotFoundErrorName } from '@web3-storage/filecoin-api/errors'
+import { parse as parseLink } from 'multiformats/link'
+import { Piece } from '@web3-storage/data-segment'
+
+/**
+ * @typedef {import('@web3-storage/filecoin-api/deal-tracker/api').DealStore} DealStore
+ * @typedef {import('./types').PieceContracts} PieceContracts
+ * @typedef {import('./types').DealArchive} DealArchive
+ * @typedef {import('../store/types').DealArchiveStore} DealArchiveStore
+ */
+
+/**
+ * On CRON tick, this function syncs deal store entries with the most up to date information stored
+ * in Spade's Oracle:
+ * - The current oracle state known is fetched, as well as the latest state from Spade endpoint.
+ * - Once both states are in memory, they are compared and a diff is generated.
+ * - Diff is stored in deal store
+ * - Handled new state of oracle is stored for comparison in next tick.
+ * 
+ * @param {object} context
+ * @param {DealStore} context.dealStore
+ * @param {DealArchiveStore} context.dealArchiveStore
+ * @param {URL} context.spadeOracleUrl
+ */
+export async function spadeOracleSyncTick ({
+  dealStore,
+  dealArchiveStore,
+  spadeOracleUrl
+}) {
+  // Get latest deal archive
+  const fetchLatestDealArchiveRes = await fetchLatestDealArchive(spadeOracleUrl)
```

Comment:
Not blocking - I imagine there will be times where the archive has not changed - would it be possible to do a head request to get an etag and avoid pulling a dump with no differences?

Then the deal archive store is more like hash => data and we store a pointer to the latest and delete (or just expire) archives we no longer need.

Reply:
I will create an issue to improve this and add a comment

Reply:
https://github.com/web3-storage/w3filecoin/issues/62